frb_example/with_flutter/rust/src/off_topic_code.rs

```diff
@@ -157,35 +157,36 @@ pub fn mandelbrot(
 
     {
         let bands = Mutex::new(pixels.chunks_mut(band_rows * bounds.0).enumerate());
+        let worker = || loop {
+            match {
+                let mut guard = bands.lock().unwrap();
+                guard.next()
+            } {
+                None => {
+                    return;
+                }
+                Some((i, band)) => {
+                    let top = band_rows * i;
+                    let height = band.len() / bounds.0;
+                    let band_bounds = (bounds.0, height);
+                    let band_upper_left = pixel_to_point(bounds, (0, top), upper_left, lower_right);
+                    let band_lower_right =
+                        pixel_to_point(bounds, (bounds.0, top + height), upper_left, lower_right);
+                    render(band, band_bounds, band_upper_left, band_lower_right);
+                }
+            }
+        };
+
+        #[cfg(not(target_family = "wasm"))]
         crossbeam::scope(|scope| {
             for _ in 0..num_threads {
-                scope.spawn(|_| loop {
-                    match {
-                        let mut guard = bands.lock().unwrap();
-                        guard.next()
-                    } {
-                        None => {
-                            return;
-                        }
-                        Some((i, band)) => {
-                            let top = band_rows * i;
-                            let height = band.len() / bounds.0;
-                            let band_bounds = (bounds.0, height);
-                            let band_upper_left =
-                                pixel_to_point(bounds, (0, top), upper_left, lower_right);
-                            let band_lower_right = pixel_to_point(
-                                bounds,
-                                (bounds.0, top + height),
-                                upper_left,
-                                lower_right,
-                            );
-                            render(band, band_bounds, band_upper_left, band_lower_right);
-                        }
-                    }
-                });
+                scope.spawn(|_| worker());
             }
         })
         .unwrap();
+
+        #[cfg(target_family = "wasm")]
```

Comment:
Just curious, crossbeam not available on wasm yet? (Surely it is ok - it is just a demo app, not core code)

Reply:
It's not, so WASM version just runs on a single worker for now. (There's the excellent [`wasm-bindgen-rayon`] library for rayon users on the web, so I think it is easier to consume as well.)

[`wasm-bindgen-rayon`]: https://lib.rs/crates/wasm-bindgen-rayon

Reply:
Sure. I use rayon as well and it is great